modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,373 @@
+	 * @param string $filtered_image The filtered image. html including img tag
+	 * @param string $context The context of the image.
+	 */
+	if ( apply_filters( 'dominant_color_img_tag_add_dominant_color', true, $attachment_id, $image_meta, $filtered_image, $context ) ) {
+		$data  = sprintf( 'data-dominantColor="%s"', $dominant_color );
+		$style = '';
+		if ( str_contains( $filtered_image, 'loading="lazy"' ) ) {
+			$style = ' style="--dominant-color: #' . $dominant_color . ';" ';
+		}
+
+		$extra_class = '';
+
+		if ( true === $image_meta['has_transparency'] ) {
+			$data       .= ' data-has-transparency="true"';
+			$extra_class = ' has-transparency ';
+		} else {
+			$data .= ' data-has-transparency="false"';
+		}
+
+		$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+
+		$extra_class   .= ( dominant_color_color_is_light( $dominant_color ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+		$filtered_image = str_replace( 'class="', 'class="' . $extra_class . ' ', $filtered_image );
+	}
+
+	return $filtered_image;
+}
+add_filter( 'wp_content_img_tag', 'dominant_color_img_tag_add_dominant_color', 20, 3 );
+
+// We don't need to use this filter anymore as the filter wp_content_img_tag is used instead.
+if ( version_compare( '6', $GLOBALS['wp_version'], '>=' ) ) {
+
+	/**
+	 * Filter the content to allow us to filter the image tags.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $content the content to filter.
+	 * @param string $context the context of the content.
+	 *
+	 * @return string content
+	 */
+	function dominant_color_filter_content_tags( $content, $context = null ) {
+		if ( null === $context ) {
+			$context = current_filter();
+		}
+
+		if ( ! preg_match_all( '/<(img)\s[^>]+>/', $content, $matches, PREG_SET_ORDER ) ) {
+			return $content;
+		}
+
+		// List of the unique `img` tags found in $content.
+		$images = array();
+
+		foreach ( $matches as $match ) {
+			list( $tag, $tag_name ) = $match;
+
+			switch ( $tag_name ) {
+				case 'img':
+					if ( preg_match( '/wp-image-([0-9]+)/i', $tag, $class_id ) ) {
+						$attachment_id = absint( $class_id[1] );
+
+						if ( $attachment_id ) {
+							// If exactly the same image tag is used more than once, overwrite it.
+							// All identical tags will be replaced later with 'str_replace()'.
+							$images[ $tag ] = $attachment_id;
+							break;
+						}
+					}
+					$images[ $tag ] = 0;
+					break;
+			}
+		}
+
+		// Reduce the array to unique attachment IDs.
+		$attachment_ids = array_unique( array_filter( array_values( $images ) ) );
+
+		if ( count( $attachment_ids ) > 1 ) {
+			/*
+			 * Warm the object cache with post and meta information for all found.
+			 * images to avoid making individual database calls.
+			 */
+			_prime_post_caches( $attachment_ids, false, true );
+		}
+
+		// Iterate through the matches in order of occurrence as it is relevant for whether or not to lazy-load.
+		foreach ( $matches as $match ) {
```

Comment:
I think we should initialize `filtered_image` before this `foreach` loop to make sure is defined.

```suggestion
                $filtered_image = null;
		foreach ( $matches as $match ) {
```